File: jaffna_vehicle_spot_java_backend/src/main/java/com/cnl/global/jaffnavicle/model/Vehicle.java
package com.cnl.global.jaffnavicle.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "vehicles")
public class Vehicle {

    @Id
    private String id;

    @Field("vehicle_name")
    private String vehicleName;

    @Indexed(unique = true)
    @Field("vehicle_number")
    private String vehicleNumber;

    @Field("model")
    private String model;

    @Field("status")
    @Builder.Default
    private String status = "Available"; // Available, Sold, In Garage

    @Field("branch_id")
    private String branchId;

    @CreatedDate
    @Field("created_at")
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Field("updated_at")
    private LocalDateTime updatedAt;
}

File: jaffna_vehicle_spot_backend/models/Vehicle.js
const mongoose = require('mongoose');

const vehicleSchema = new mongoose.Schema({
  vehicle_name: { type: String, required: true },
  vehicle_number: { type: String, required: true, unique: true },
  model: { type: String, required: true },
  status: { type: String, enum: ['Available', 'Sold', 'In Garage'], default: 'Available' },
  branch_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Branch', required: true },
}, {
  timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
});

module.exports = mongoose.model('Vehicle', vehicleSchema);
